File: src/AILogistics.Domain/Entities/Shipment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AILogistics.Domain.Entities
{
    public enum PaymentStatus
    {
        Pending = 0,
        Paid = 1,
        Failed = 2
    }

    public enum ShipmentStatus
    {
        Pending = 0,
        PickedUp = 1,
        InTransit = 2,
        OutForDelivery = 3,
        Delivered = 4,
        Cancelled = 5,
        Returned = 6
    }

    public enum ShipmentPriority 
    { 
        Normal = 0,
        High = 1,
        Urgent = 2
    }

    public class Shipment : BaseEntity
    {

        public string ShipmentNumber { get; set; }
        public int CustomerId { get; set; }
        public string Origin { get; set; }
        public string Destination { get; set; }

        public decimal WeightKg { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public ShipmentStatus Status { get; set; }
        public DateTime PickupDate { get; set; }
        public DateTime DeliveryDate { get; set; }
        public ShipmentPriority Priority { get; set; }
        public decimal DistanceKm { get; set; }

        public Customer Customer { get; set; } = null!;

        public ICollection<TrackingEvent> TrackingEvents { get; set; } =  new List<TrackingEvent>();

    }
}

File: Client/src/types/shipment.ts
export interface Shipment {
    id: number;
    shipmentNumber: string;
    customerId: number;
    customerName: string;
    origin: string;
    destination: string;
    weightKg: number;
    shipmentStatus: ShipmentStatus;
    paymentStatus: PaymentStatus;
    pickupDate: string;
    deliveryDate: string;
    createdAt: string;
    updatedAt: string;
    rowVersion: string;
}

export interface ShipmentEditableFields {
    origin: string;
    destination: string;
    weightKg: number;
    pickupDate: string;
    deliveryDate: string;
}

export interface CreateShipmentRequest extends ShipmentEditableFields {
    customerId: number;
}

export interface UpdateShipmentRequest extends ShipmentEditableFields {
    rowVersion: string;
}

export const PaymentStatus = {
    Pending : 0,
    Paid : 1,
    Failed : 2,
} as const;

export type PaymentStatus = (typeof PaymentStatus)[keyof typeof PaymentStatus];

export const ShipmentStatus = {
    Pending: 0,
    PickedUp: 1,
    InTransit: 2,
    OutForDelivery: 3,
    Delivered: 4,
    Cancelled: 5,
    Returned: 6,
} as const;

export type ShipmentStatus = (typeof ShipmentStatus)[keyof typeof ShipmentStatus];
